Replay.playground/Contents.swift

```diff
@@ -0,0 +1,119 @@
+import Combine
+import Foundation
+
+var subscriptions: Set<AnyCancellable> = []
+
+public extension Publishers {
+    final class Replay<Upstream: Publisher>: Publisher {
+        public typealias Output = Upstream.Output
+        public typealias Failure = Upstream.Failure
+        
+        private var subscription: AnyCancellable?
+        private let upstream: AnyPublisher<Upstream.Output, Upstream.Failure>
+        private var output: [Upstream.Output] = []
+        private var completion: Subscribers.Completion<Upstream.Failure>?
+        
+        init(upstream: Upstream) {
+            self.upstream = upstream.share().eraseToAnyPublisher()
+            subscription = self.upstream.sink(
+                receiveCompletion: { [weak self] completion in self?.completion = completion },
+                receiveValue: { [weak self] value in self?.output.append(value) })
+        }
+        
+        public func receive<S>(subscriber: S) where S : Subscriber, Failure == S.Failure, Output == S.Input {
+            let subscription = ReplaySubscription(subscriber: subscriber, upstream: upstream, output: output, completion: completion)
+            subscriber.receive(subscription: subscription)
+        }
+        
+        deinit {
+            subscription = nil
+        }
+    }
+}
+
+private extension Publishers.Replay {
+    private class ReplaySubscription<S: Subscriber>: Subscriber, Subscription where S.Failure == Upstream.Failure, S.Input == Upstream.Output {
+        typealias Input = Upstream.Output
+        typealias Failure = Upstream.Failure
+        
+        private let subscriber: S
+        private var demand: Subscribers.Demand = .none
+        private var output: [Upstream.Output] = []
+        private var completion: Subscribers.Completion<Upstream.Failure>?
+        private var upstream: AnyPublisher<Upstream.Output, Upstream.Failure>
+        private var upstreamSubscription: Subscription?
+        
+        init(subscriber: S, upstream: AnyPublisher<Upstream.Output, Upstream.Failure>, output: [Upstream.Output], completion: Subscribers.Completion<Upstream.Failure>?) {
```

Comment:
Yeah definitely it works but the whole process is over complicated.
Also it does not have a counterpart with RxSwift where a buffer size is passed as input and it's a `Connectable`.
I think I finally found a nice solution with about 30 lines 😱
Backpressure is not a priority for this, and in case required a subscription that will take care of it would be easy to introduce.
Please have look on the updated #1
Keep in mind that this is `Replay`, which share the subscription and is a connectable.
`ShareReplay` would be just the wrapped `Replay` using autoconnect.